### PR TITLE
Support SDL color

### PR DESCRIPTION
**Pull request by Vraiment**
Hey, this is my first contribution to a project ever.

I saw you have issue #55 open and I realized it was something I could do, please let me know if I missed something, I tried to follow the style from Point and Rect classes.

### CODE REVIEW

File: SDL2pp/Texture.hh [hunk 1]
```diff
@@ -422,6 +423,20 @@ public:
...
+	/// \param[in] color Color to be used when multiplied into render copy operations
+	///
+	/// \returns Reference to self
+	///
+	/// \throws SDL2pp::Exception
+	///
+	/// \see http://wiki.libsdl.org/SDL_SetTextureColorMod
+	///
+	////////////////////////////////////////////////////////////
+	Texture& SetColorAndAlphaMod(const Color color = Color{255, 255, 255, 255});
```
Comment (AMDmi3) [inline SDL2pp/Texture.hh]: I liked SDL_ALPHA_OPAQUE you've used above, let's use it here too.

File: SDL2pp/Texture.hh [hunk 2]
```diff
@@ -536,6 +551,18 @@ public:
...
 	void GetColorMod(Uint8& r, Uint8& g, Uint8 &b) const;
+
+	////////////////////////////////////////////////////////////
+	/// \brief Get the additional color value multiplied into render copy operations
+	///
+	/// \return Color object with the values used to do render copy operations
+	///
+	/// \throws SDL2pp::Exception
+	///
+	/// \see http://wiki.libsdl.org/SDL_GetTextureColorMod
```
Comment (AMDmi3) [inline SDL2pp/Texture.hh]: Could reference SDL_GetTextureAlphaMod here too

File: SDL2pp/Texture.hh [hunk 3]
```diff
@@ -422,6 +423,20 @@ public:
...
 	////////////////////////////////////////////////////////////
+	/// \brief Set an additional color value multiplied into render copy operations
+	///
+	/// \param[in] color Color to be used when multiplied into render copy operations
+	///
+	/// \returns Reference to self
+	///
+	/// \throws SDL2pp::Exception
+	///
+	/// \see http://wiki.libsdl.org/SDL_SetTextureColorMod
```
Comment (AMDmi3) [inline SDL2pp/Texture.hh]: Could reference SDL_SetTextureAlphaMod here too

File: SDL2pp/Surface.hh [hunk 4]
```diff
@@ -478,6 +491,20 @@ public:
...
 	////////////////////////////////////////////////////////////
+	/// \brief Set an additional color value multiplied into blit operations
+	///
+	/// \param[in] color Color to be multiplied into blit operations
+	///
+	/// \returns Reference to self
+	///
+	/// \throws SDL2pp::Exception
+	///
+	/// \see http://wiki.libsdl.org/SDL_SetSurfaceColorMod
```
Comment (AMDmi3) [inline SDL2pp/Surface.hh]: Could reference SDL_SetSurfaceAlphaMod here too

File: SDL2pp/Surface.hh [hunk 5]
```diff
@@ -393,6 +394,18 @@ public:
 	////////////////////////////////////////////////////////////
 	/// \brief Get the additional color value multiplied into blit operations
 	///
+	/// \return Color object with the values used to do blit operations
+	///
+	/// \throws SDL2pp::Exception
+	///
+	/// \see http://wiki.libsdl.org/SDL_GetSurfaceColorMod
```
Comment (AMDmi3) [inline SDL2pp/Surface.hh]: Could reference SDL_GetSurfaceAlphaMod here too

File: SDL2pp/Renderer.hh [hunk 6]
```diff
@@ -302,6 +303,20 @@ public:
...
+	/// \param[in] color Color to draw on the rendering target
+	///
+	/// \returns Reference to self
+	///
+	/// \throws SDL2pp::Exception
+	///
+	/// \see http://wiki.libsdl.org/SDL_SetRenderDrawColor
+	///
+	////////////////////////////////////////////////////////////
+	Renderer& SetDrawColor(const Color color);
```
Comment (AMDmi3) [inline SDL2pp/Renderer.hh]: ~~const~~

**Comment by AMDmi3** [comment]
Also please ignore appveyor failure, it's unrelated.

**Comment by Vraiment** [comment]
As for the tests, I don't get them very well how they work. I guess they are not unit tests but rather like integration tests, right?

Have you seen [Catch](https://github.com/philsquared/Catch)? Is an awesome very simple and elegant testing framework for C++

File: SDL2pp/Color.cc [hunk 7]
```diff
@@ -1,6 +1,6 @@
 /*
   libSDL2pp - C++11 bindings/wrapper for SDL2
-  Copyright (C) 2013-2015 Dmitry Marakasov <amdmi3@amdmi3.ru>
```
Comment (AMDmi3) [inline SDL2pp/Color.cc]: Please set correct date as well.

File: SDL2pp/Renderer.cc [hunk 8]
```diff
@@ -197,7 +197,7 @@ Renderer& Renderer::SetDrawColor(Uint8 r, Uint8 g, Uint8 b, Uint8 a) {
 	return *this;
 }
 
-Renderer& Renderer::SetDrawColor(const Color color) {
```
Comment (AMDmi3) [inline SDL2pp/Renderer.cc]: Sorry, I've misguided you. It should better be ```const Color&```. Before, I did benchmarks for ```Rect``` vs. ```const Rect&``` arguments, and the latter was better. We should do the same way here.

**Comment by AMDmi3** [comment]
> As for the tests, I don't get them very well how they work. I guess they are not unit tests but rather like integration tests, right?

There are both.

> Have you seen Catch? Is an awesome very simple and elegant testing framework for C++

No, but I've heard of it. From what I've heard, I'm using the very similar homegrown framework (grep ```testing.h```)

**Comment by Vraiment** [comment]
There, sorry for the mess, this should be good to go

**Comment by AMDmi3** [comment]
Looks good now apart from the Doxyfile.in changes. Since these were merged in separate PR they should go awat if you just rebase onto master.

**Comment by AMDmi3** [comment]
:+1::+1::+1:

**Pull request merged**